### PR TITLE
[MM-11222] Migrate 'Channel.GetMemberCount' to Sync by default

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR will migrate `Channel.GetMemberCount` to Sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11222

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1643,31 +1643,29 @@ func (s SqlChannelStore) GetMemberCountFromCache(channelId string) int64 {
...
-		if s.metrics != nil {
-			s.metrics.IncrementMemCacheMissCounter("Channel Member Counts")
-		}
+	if s.metrics != nil {
+		s.metrics.IncrementMemCacheMissCounter("Channel Member Counts")
+	}
 
-		count, err := s.GetReplica().SelectInt(`
+	count, err := s.GetReplica().SelectInt(`
 			SELECT
```
Comment (jespino) [inline store/sqlstore/channel_store.go]: Nit: can you unindent the query string lines consistently?

File: app/channel.go [hunk 2]
```diff
@@ -1479,7 +1475,13 @@ func (a *App) LeaveChannel(channelId string, userId string) *model.AppError {
 		uc <- store.StoreResult{Data: user, Err: err}
 		close(uc)
 	}()
-	ccm := a.Srv.Store.Channel().GetMemberCount(channelId, false)
+
+	cc := make(chan store.StoreResult, 1)
+	go func() {
```
Comment (jespino) [inline app/channel.go]: Why cc instead of ccm?

Comment (iomodo) [reply app/channel.go]: cc = countChan
may be better mcc = memberCountChan ?
I did not understand ccm, is it "chan for counting members" or "count channel members"?

Comment (jespino) [reply app/channel.go]: Yes, for sure wasn't very consistent, I like the `mcc` option.

**Pull request merged**